### ISSUE TITLE
java GUI not working

### ISSUE THREAD

**Issue by jianshu93** [molbiodiv/bcgTree#50]
Dear bacgtree team,

Thanks for the GUI version. It does not work however after running java -jar ./bcgTree.jar:

Java HotSpot(TM) 64-Bit Server VM warning: CodeCache is full. Compiler has been disabled.
Java HotSpot(TM) 64-Bit Server VM warning: Try increasing the code cache size using -XX:ReservedCodeCacheSize=
CodeCache: size=131072Kb used=7200Kb max_used=7203Kb free=123871Kb
 bounds [0x00000001049f8000, 0x0000000105108000, 0x000000010c9f8000]
 total_blobs=2639 nmethods=2073 adapters=480
 compilation: disabled (not enough contiguous free space left)
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at bcgTree.gui.BcgTree.renameProteomes(BcgTree.java:547)
	at bcgTree.gui.BcgTree$1.actionPerformed(BcgTree.java:313)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:262)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3318)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at bcgTree.gui.BcgTree.renameProteomes(BcgTree.java:547)
	at bcgTree.gui.BcgTree$1.actionPerformed(BcgTree.java:313)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:262)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3318)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

Any idea? I am using the most recent java and JRE from oracle.

Thanks,

Jianshu

**Comment by iimog** [comment]
Does this happen right away when you run `java -jar ./bcgTree.jar`? Or does the GUI open, and you click "Run"? In the latter case, make sure, to provide some proteomes before you start. It looks like you are trying to run bcgTree without providing any input.

**Comment by jianshu93** [comment]
I did provide everything and click run. Will share a screenshot soon.

Thanks，

Jianshu

**Comment by jianshu93** [comment]
Hello @iimog ,

I attached the screen shot when open the window and also after click run. All denpendencies are there.
<img width="1204" alt="Screenshot 2023-12-19 at 7 39 56 PM" src="https://github.com/molbiodiv/bcgTree/assets/38149286/883e38d9-7afd-40a8-b7be-7465d9af19b8">

<img width="1204" alt="Screenshot 2023-12-19 at 7 41 52 PM" src="https://github.com/molbiodiv/bcgTree/assets/38149286/a29965ac-7644-4c09-a616-dc0057d334c8">

**Comment by iimog** [comment]
Thank you for the screenshots. I'm able to reproduce the error now. It seems to happen when all inputs are genomes and there is no proteome file. Of course, bcgTree should work in this case as well (and it does, it is just the GUI that has problems, but also the GUI should work). I'll fix this problem. But I'm currently on vacation, so I won't be able to fix it this year. 
A workaround for now is to translate at least one genome file manually and provide it as proteome. The command that bcgTree uses for the translation is:
```bash
prodigal -i final_contigs_AM-654-A11.fasta -o final_contigs_AM-654-A11.fasta.prodigal.genes -a final_contigs_AM-654-A11.fasta.prodigal.faa
```
And the file `final_contigs_AM-654-A11.fasta.prodigal.faa` is the proteome that is used further.
```

**Comment by jianshu93** [comment]
Hello @iimog,

Ok no problem, I will use proteome first. Can you also please add fasttree support.  I understand that RAxML is the most accurate one but too slow for more than 30 genomes or so, for a GUI application, most use cases are personal computers like laptops, which will be slow.  fasttree can be 10 times faster while provide almost identical result with RAxML, very useful for quick check of phylogeny of some genomes.

Thanks,

Jianshu

**Comment by jianshu93** [comment]
Hello @iimog,

It seems with the newest version of RAXML, v8.2.12, there are alway problems with respect to partition file:


RAxML was called as follows:

raxmlHPC-PTHREADS -f a -m GTRGAMMA -p 740281 -q /Users/jianshuzhao/Github/bcgTree/data/SAR11_SAG_try_bcgtree_full_alignment.concat.partition -s /Users/jianshuzhao/Github/bcgTree/data/SAR11_SAG_try_bcgtree_full_alignment.concat.fa -w /Users/jianshuzhao/Github/bcgTree/data/ -n SAR11_SAG_bcgtree_final -T 4 -x 28840 -N 100 


Partition PF00276.16 number 1 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition PF00281.15 number 2 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00297.18 number 3 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition PF00410.15 number 7 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00416.18 number 9 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00466.16 number 10 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF01025.15 number 13 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00001 number 15 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00002 number 16 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00009 number 17 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00012 number 18 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00029 number 20 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00059 number 22 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00060 number 23 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00061 number 24 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00062 number 25 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00082 number 27 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00115 number 30 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00152 number 32 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00158 number 33 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00165 number 34 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00168 number 36 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00442 number 50 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00663 number 59 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00810 number 61 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00855 number 62 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00952 number 64 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00959 number 65 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00964 number 67 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00981 number 69 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01021 number 73 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01024 number 74 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01031 number 76 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR01032 number 77 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01044 number 78 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01049 number 79 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR01050 number 80 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01066 number 82 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01067 number 83 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01079 number 85 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01169 number 87 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01632 number 91 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00276.16 number 1 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition PF00281.15 number 2 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00297.18 number 3 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition PF00410.15 number 7 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00416.18 number 9 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF00466.16 number 10 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition PF01025.15 number 13 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00001 number 15 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00002 number 16 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00009 number 17 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00012 number 18 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00029 number 20 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00059 number 22 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00060 number 23 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00061 number 24 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00062 number 25 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00082 number 27 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00115 number 30 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00152 number 32 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00158 number 33 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00165 number 34 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00168 number 36 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00442 number 50 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00663 number 59 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00810 number 61 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00855 number 62 has a problem, the number of expected states is 20 the number of states that are present is 17.
Please go and fix your data!

Partition TIGR00952 number 64 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00959 number 65 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR00964 number 67 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR00981 number 69 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01021 number 73 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01024 number 74 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01031 number 76 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR01032 number 77 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01044 number 78 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01049 number 79 has a problem, the number of expected states is 20 the number of states that are present is 18.
Please go and fix your data!

Partition TIGR01050 number 80 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01066 number 82 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01067 number 83 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01079 number 85 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01169 number 87 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Partition TIGR01632 number 91 has a problem, the number of expected states is 20 the number of states that are present is 19.
Please go and fix your data!

Segmentation fault: 11




I attached the alignment file and partition file from the java GUI:

[Archive.zip](https://github.com/molbiodiv/bcgTree/files/13773560/Archive.zip)



Any idea?

Thanks

**Comment by jianshu93** [comment]
hello @iimog,

I am new to Java but I have a similar pipeline that I want to put together as bcgtreeGUI. It would be nice if you could share the Java code to make the .jar file. 

Thank you,

Jianshu

**Comment by iimog** [comment]
The GUI issue is fixed in v1.2.1

**Comment by jianshu93** [comment]
Thanks for the quick fix and I tested everything. It worked! Again for only 5 genomes, it takes more than 20 minutes with 2 threading. It seems RAxML threading is not so efficient, the same 5 genome for iqtree2, it takes 1 minutes with 2 threads.
